packages/react/src/components/Table/Table.tsx

```diff
@@ -1,21 +1,108 @@
-import React, { TableHTMLAttributes } from 'react';
+import React, { useMemo } from 'react';
 import classNames from 'classnames';
+import { TableProvider } from './TableContext';
 
-interface TableProps extends TableHTMLAttributes<HTMLTableElement> {
+export type Column = {
+  align: ColumnAlignment;
+  width?: ColumnWidth;
+};
+export type ColumnAlignment = 'start' | 'center' | 'end';
+export type ColumnWidth =
+  | 'auto'
+  | 'min-content'
+  | 'max-content'
+  | `${number}`
+  | `${number}%`
+  | `${number}fr`;
+export type RowAlignment = 'start' | 'center';
+
+type TableBaseProps = {
+  layout: never;
+  columns: never;
+  variant?: 'border';
+};
+
+type TableGridProps = {
+  layout: 'grid';
+  columns?: Array<Column> | number;
   variant?: 'border';
-}
-
-const Table = ({ children, className, variant, ...other }: TableProps) => (
-  <table
-    className={classNames(
-      'Table',
-      variant === 'border' && 'Table--border',
-      className
-    )}
-    {...other}
-  >
-    {children}
-  </table>
+};
+
+type TableProps = (TableBaseProps | Partial<TableGridProps>) &
+  React.TableHTMLAttributes<HTMLTableElement>;
+
+const Table = React.forwardRef<HTMLTableElement, TableProps>(
+  (
+    {
+      children,
+      className,
+      variant,
+      layout,
+      columns: columnsProp = [],
+      style,
+      ...other
+    },
+    ref
+  ) => {
+    const isGridLayout = layout === 'grid';
+    const columns: Column[] = useMemo(() => {
+      if (typeof columnsProp === 'number') {
+        return columnsProp > 0
+          ? Array(columnsProp).fill({ align: 'start' })
+          : [{ align: 'start' }];
+      }
+
+      return columnsProp;
+    }, [columnsProp]);
+
+    const styleTemplateColumns = useMemo(() => {
+      if (layout !== 'grid') {
+        return;
+      }
+
+      if (!columns) {
+        return 'auto';
+      }
+
+      return columns
+        .map(({ width }) => {
+          if (!width) {
+            return 'auto';
+          } else {
+            return width;
+          }
+        })
```

Comment:
Since it appears that width is always a string (if set) we don't need to account for the pesky falsey `0` (number), can't we simplify this? 

```suggestion
        .map(({ width }) => width || 'auto')
```

Reply:
(this is a nitpick)